app/src/main/java/com/example/househomey/AddItemFragment.java

```diff
@@ -51,39 +60,82 @@ public AddItemFragment(CollectionReference itemRef) {
+            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+            }
+
+            @Override
+            public void afterTextChanged(Editable editable) {
+                if (editable.equals(descView.getEditableText())) {
+                    isRequiredFieldEmpty(R.id.add_item_description_layout);
+                } else if (editable == costView.getEditableText()) {
+                    isRequiredFieldEmpty(R.id.add_item_cost_layout);
+                }
+            }
+        };
+        descView.addTextChangedListener(emptyTextWatcher);
+        costView.addTextChangedListener(emptyTextWatcher);
 
         // Add listener for confirm and back buttons
         rootView.findViewById(R.id.add_item_confirm_button).setOnClickListener(v -> addItem());
-        rootView.findViewById(R.id.add_item_back_button).setOnClickListener(v -> returnToHomeScreen());
+        rootView.findViewById(R.id.add_item_back_button).setOnClickListener(v -> navigateToFragmentPage((AppCompatActivity) getContext(), new HomeFragment(itemRef)));
 
         return rootView;
     }
 
     /**
-     * This shows a datePicker for the user to select the date of the item acquisition
+     * Checks whether the text input for a required form field is empty.
+     * Additionally, if empty, it sets an inline error message on the input field.
+     * If not empty, it removes the inline error.
+     *
+     * @param id Id of the required field's TextInputLayout
+     * @return a boolean indicating if the field is empty
      */
-    private void showDatePicker() {
-        MaterialDatePicker<Long> datePicker = MaterialDatePicker.Builder.datePicker()
-                .setInputMode(MaterialDatePicker.INPUT_MODE_CALENDAR)
-                .build();
-
-        datePicker.addOnPositiveButtonClickListener(selection -> {
-            dateAcquired = new Date(selection);
-            dateTextView.setText(datePicker.getHeaderText());
-        });
-
-        datePicker.show(getParentFragmentManager(), "Date Picker");
+    private boolean isRequiredFieldEmpty(int id) {
+        TextInputLayout textInputLayout = (TextInputLayout) getView().findViewById(id);
+        if (TextUtils.isEmpty(textInputLayout.getEditText().getText().toString().trim())) {
+            textInputLayout.setError("This field is required");
+            return true;
+        }
+        textInputLayout.setError(null);
+        return false;
     }
 
     /**
-     * Adds the user input data to the firestore database
+     * Adds the user input data to a new item in their Firestore collection
      */
     private void addItem() {
-        // TODO: add input validation
+        // Check that required fields are filled before submitting
+        boolean invalidDesc = isRequiredFieldEmpty(R.id.add_item_description_layout);
+        boolean invalidDate = isRequiredFieldEmpty(R.id.add_item_date_layout);
+        boolean invalidCost = isRequiredFieldEmpty(R.id.add_item_cost_layout);
```

Comment:
This one's not entirely your fault... I didn't set a rounding mode for the big decimal, turns out we should. Please pu this on line 39 in Item.java ```this.cost = new BigDecimal((String)  objects.requireNonNull(data.get("cost"))).setScale(2, RoundingMode.HALF_DOWN);```

If you enter with more than two decimal places it breaks the program. So we should prevent the user from entering more than two decimal places in the cost field in the first place, probably regex is the easiest solution here to stop that.
```"^\\$?\\d*\\.?\\d{0,2}$"```

Reply:
@ldbonkowski just to confirm. Should we use `RoundingMode.HALF_DOWN` or `RoundingMode.HALF_UP`?

I'm adding this line to Item, as well as using a listener+BigDecimal to fix the allowed input. They can still input more than 2 decimals, but once they go off the input, it will replace with the rounded version

Reply:
HALF_UP makes more sense, my bad. Sounds good.